pycloudlib/openstack/tests/test_cloud.py

```diff
@@ -0,0 +1,40 @@
+"""Openstack cloud tests."""
+from unittest import mock
+
+import pytest
+from openstack.connection import Connection
+
+from pycloudlib.openstack.cloud import Openstack
+
+
+@mock.patch('pycloudlib.key.KeyPair.public_key_content',
+            new_callable=mock.PropertyMock,
+            return_value='pretend public key')
+@mock.patch('openstack.connect', spec=Connection)
+class TestOpenstackKeypair:
+    """Tests covering _get_openstack_keypair."""
+
+    def test_keypair_doesnt_exist(self, m_openstack, m_keypair):
+        """Test no pre-existing openstack keypair."""
+        m_openstack.return_value.get_keypair.return_value = None
+        cloud = Openstack(tag='test', network=None)
+        cloud._get_openstack_keypair()
+        assert 1 == cloud.conn.create_keypair.call_count
+
+    def test_keypairs_match(self, m_openstack, m_keypair):
```

Comment:
`m_keypair` threw me off here: I assumed that would be a mock of `pycloudlib.key.KeyPair` (making this test very confusing :p). Perhaps `m_public_key_content`? Still a little confusing, given it's a property mock (whereas you'd never call `public_key_content` of course).